src/Compilers/CSharp/Portable/Symbols/Source/SourceEventSymbol.cs

```diff
@@ -194,8 +203,26 @@ protected abstract AttributeLocation AllowedAttributeLocations
         /// </remarks>
         private CustomAttributesBag<CSharpAttributeData> GetAttributesBag()
         {
-            if ((_lazyCustomAttributesBag == null || !_lazyCustomAttributesBag.IsSealed) &&
-                LoadAndValidateAttributes(OneOrMany.Create(this.AttributeDeclarationSyntaxList), ref _lazyCustomAttributesBag))
+            var bag = _lazyCustomAttributesBag;
+            if (bag != null && bag.IsSealed)
+            {
+                return bag;
+            }
+
+            bool bagCreatedOnThisThread;
+
+            if (SourcePartialDefinitionPart is { } definitionPart)
+            {
+                Debug.Assert(!ReferenceEquals(definitionPart, this));
+                bag = definitionPart.GetAttributesBag();
+                bagCreatedOnThisThread = Interlocked.CompareExchange(ref _lazyCustomAttributesBag, bag, null) == null;
+            }
+            else
+            {
+                bagCreatedOnThisThread = LoadAndValidateAttributes(this.GetAttributeDeclarations(), ref _lazyCustomAttributesBag);
+            }
+
+            if (bagCreatedOnThisThread)
```

Comment:
I found the original code a bit confusing.. we thought that we would *never* race to bind attributes on an event symbol? What?

Reply:
I think the original code was fine - it's equivalent to the current `else` branch, i.e., if `SourcePartialDefinitionPart is { } definitionPart` was never true (which it wasn't previously of course). It was just packed into lot less lines.

Reply:
Ah, I get it now. I didn't understand what the return value of `LoadAndValidateAttributes` meant.